ops/main.py

```diff
@@ -42,6 +43,87 @@
 logger = logging.getLogger()
 
 
+@dataclasses.dataclass(frozen=True)
+class _JujuContext:
+    # source: https://juju.is/docs/juju/charm-environment-variables
+    # the HookVars function: https://github.com/juju/juju/blob/3.6/worker/uniter/runner/context/context.go#L1398
+    # only allow accesing these vars because these are the currently used vars in main.py
+
+    juju_action: '_JujuAction'
+    juju_charm_dir: Optional[str]
+    juju_debug: Optional[str]
+    juju_departing_unit: str
+    juju_dispatch_path: Optional[str]
+    juju_notice: '_JujuNotice'
+    juju_pebble_check_name: str
+    juju_relation: '_JujuRelation'
+    juju_remote_app: str
+    juju_remote_unit: str
+    juju_secret: '_JujuSecret'
+    juju_storage_id: str
```

Comment:
🤔 this field is called "storage tag" in juju codebase. The change is from 3 years ago, maybe we can follow that lead? wdyt?